test/test_framework.py

```diff
@@ -1628,33 +1692,35 @@ def callback_method(self, event: ops.EventBase):
         self.called = True
 
 
-@patch('sys.stderr', new_callable=io.StringIO)
-class BreakpointTests(BaseTestCase):
+class TestBreakpoint:
 
-    def test_ignored(self, fake_stderr: io.StringIO):
+    def test_ignored(
+        self,
+        request: pytest.FixtureRequest,
+        caplog: pytest.LogCaptureFixture,
+    ):
+        fake_stderr = io.StringIO()
         # It doesn't do anything really unless proper environment is there.
         with patch.dict(os.environ):
             os.environ.pop('JUJU_DEBUG_AT', None)
-            framework = self.create_framework()
+            framework = create_framework(request)
 
         with patch('pdb.Pdb.set_trace') as mock:
             # We want to verify that there are *no* logs at warning level.
             # However, assertNoLogs is Python 3.10+.
-            try:
-                with self.assertLogs(level="WARNING"):
-                    framework.breakpoint()
-            except AssertionError:
-                pass
-            else:
-                self.fail("No warning logs should be generated")
+
+            framework.breakpoint()
+            warning_logs = [
+                record for record in caplog.records if record.levelno == logging.WARNING]
+            assert len(warning_logs) == 0, "No warning level logs were generated."
```

Comment:
This message seems inverted (it's printed if the assertion fails). However, I think we can probably just drop it, as it doesn't really add any value over what the assertion itself says.